service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseService.java

```diff
@@ -0,0 +1,137 @@
+    @Autowired
+    private DocumentUploadClientApi documentUploadClient;
+    @Autowired
+    private CoreCaseDataApi coreCaseDataApi;
+    @Autowired
+    private AuthTokenGenerator authTokenGenerator;
+
+    /**
+     *
+     * @param authorization
+     * @param userId
+     * @param request
+     * @throws JSONException
+     * @throws IOException
+     *
+     * Converts request @param to a map that contains the case data and the headers that can be used to create the pdf and view contents of the
+     * callback
+     */
+    @Async
+    @SuppressWarnings("unchecked")
+    public void handleCaseSubmission(
+        String authorization,
+        String userId,
+        CallbackRequest request) throws JSONException, IOException {
+        byte[] template = documentTemplates.getHtmlTemplate();
+
+        ObjectMapper mapper = new ObjectMapper();
+        Map<String, Object> map = mapper.convertValue(request.getCaseDetails(), Map.class);
+
+        System.out.println("map = " + map);
+        byte[] pdfDocument = converter.convert(template, map);
+
+        Document document = uploadDocument(userId, authorization, authTokenGenerator.generate(), pdfDocument, getFileName(request.getCaseDetails()));
+
+        System.out.println("binary = " + document.links.binary.href);
+        System.out.println("self = " + document.links.self.href);
+
+        String caseId = request.getCaseDetails().getId().toString();
+
+        StartEventResponse startEventResponse = coreCaseDataApi.startEventForCaseWorker(authorization, authTokenGenerator.generate(), userId, JURISDICTION_ID,
+            CASE_TYPE, caseId,"PDF");
+
+        System.out.println("startEventResponse = " + startEventResponse);
+        System.out.println("startEventResponse.token = " + startEventResponse.getToken());
+
+        Map<String, Object> data = Maps.newHashMap();
+
+        Map<String, Object> documentData = Maps.newHashMap();
+        documentData.put("document_url", document.links.self.href.replace("dm-store:8080", "localhost:3453"));
+        documentData.put("document_binary_url", document.links.binary.href.replace("dm-store:8080", "localhost:3453"));
+        documentData.put("document_filename", document.originalDocumentName);
+        data.put("submittedForm", documentData);
+
+        CaseDataContent body = CaseDataContent.builder()
+            .eventToken(startEventResponse.getToken())
+            .event(Event.builder()
+                .id(startEventResponse.getEventId())
+                .summary("Attach submitted form")
+                .description("Attach submitted form")
+                .build())
+            .data(data)
+            .build();
+
+        CaseDetails caseDetails = coreCaseDataApi.submitEventForCaseWorker(authorization,
+            authTokenGenerator.generate(), userId, JURISDICTION_ID, CASE_TYPE, caseId, false, body);
+    }
+
+    private Document uploadDocument(String userId, String authorization,
```

Comment:
Please create another service class that uploads document to DM. That will make our life easier in terms of testing it.